pkg/client/selfhosted/path.go

```diff
@@ -0,0 +1,42 @@
+package selfhosted
+
+import (
+	"errors"
+	"fmt"
+	"net/url"
+	"regexp"
+	"strings"
+
+	log "github.com/sirupsen/logrus"
+)
+
+var (
+	regTemplate = (`(^(.*\.)?%s$)`)
+)
+
+func (c *Client) IsHost(host string) bool {
+
+	u, err := url.Parse(c.Options.URL)
+	if err != nil {
+		// If we can't parse the host given by the options, it's useless to keep running
+		log.Fatalf("failed parsing host: %s", c.Options.URL)
+		return false
+	}
+
+	urlRegex := fmt.Sprintf(regTemplate, u.Host)
+	reg := regexp.MustCompile(urlRegex)
```

Comment:
This can all be moved to `selfhosted.New`. This means we exit early if there is an issue, and we don't have the fatal.

This should leave nothing but `return reg.MatchString(host)`